ast/desugar/Desugar.cc

```diff
@@ -929,19 +944,28 @@ ExpressionPtr node2TreeImplBody(DesugarContext dctx, parser::Node *what) {
                     DuplicateHashKeyCheck::checkSendArgs(dctx, numPosArgs, args);
 
                     ExpressionPtr res;
-                    if (block == nullptr) {
+                    if (blockPassArg == nullptr) {
+                        // Desugar any call without a block pass argument. Block literals are handled here.
+                        // E.g. `a.map { |x| x.to_s }
                         res = MK::Send(loc, move(rec), send->method, send->methodLoc, numPosArgs, move(args), flags);
                     } else {
-                        if (auto lit = cast_tree<Literal>(block); lit && lit->isSymbol()) {
+                        if (auto lit = cast_tree<Literal>(blockPassArg); lit && lit->isSymbol()) {
+                            // Desugar a call with a Symbol block pass argument, like `a.map(&:to_s)`
+
+                            auto desugaredBlockLiteral = symbol2Proc(dctx, move(blockPassArg));
+                            args.emplace_back(move(desugaredBlockLiteral));
+                            flags.hasBlock = true;
+
                             res =
                                 MK::Send(loc, move(rec), send->method, send->methodLoc, numPosArgs, move(args), flags);
-                            ast::cast_tree_nonnull<ast::Send>(res).setBlock(symbol2Proc(dctx, move(block)));
                         } else {
+                            // Desugar a call with any other expression as a block pass argument,
+                            // like `a.map(&:method(:foo))`
```

Comment:
```suggestion
                            // like `a.map(&blk)`
```

I think the fact that there's still a Symbol literal in the code snippet makes it harder to skim which case this is.

If you really want the `method(:foo)` expression as your example, we at least need to remove the colon after the `&`: `&method(:foo)` not `&:method(:foo)`